sirepo/auth_role_moderation.py

```diff
@@ -22,11 +22,81 @@
 
 _cfg = None
 
+
+@sirepo.api_perm.require_adm
+def api_admModerate():
+
+    def _send_moderation_status_email(info):
+        approved_denied_subject = f'{info.app_name} Access Request '
+        email_subject = PKDict(
+            approve=approved_denied_subject + 'Approved',
+            clarify=f'Sirepo {info.app_name}: Additional Info?',
+            deny=approved_denied_subject + 'Denied',
+        )
+        if sirepo.auth.dev_and_no_method('email'):
```

Comment:
I tried removing it, but I ran into this error.

```
Traceback (most recent call last):
  File "/home/vagrant/src/radiasoft/sirepo/sirepo/uri_router.py", line 77, in call_api
    r = flask.make_response(f(**kwargs) if kwargs else f())
  File "/home/vagrant/src/radiasoft/sirepo/sirepo/auth_role_moderation.py", line 86, in api_admModerate
    _send_moderation_status_email(p)
  File "/home/vagrant/src/radiasoft/sirepo/sirepo/auth_role_moderation.py", line 44, in _send_moderation_status_email
    recipient=sirepo.auth.user_name(info.uid),
  File "/home/vagrant/src/radiasoft/sirepo/sirepo/auth/__init__.py", line 519, in user_name
    'UserModel',
AttributeError: module 'sirepo.auth.guest' has no attribute 'UserModel'
```

What should we do for guest users since they have no UserModel?

Reply:
We should never get a moderation request from a guest user so this code should never be executed in that case. How is a guest user asking to moderate?

Reply:
There is nothing that prevents a guest user from asking to moderate. [This test](https://github.com/radiasoft/sirepo/blob/8cb1f59751554819c468bf4cdd36601c23a8637e/tests/role_moderation_test.py#L19) shows an example of it. Another way to show it is to set a sim_type as a moderated sim_type, log in as a guest user, delete all roles for the guest user, and then go to that sim_type. The guest user will be prompted for a moderation reason. 

How should guest users be prevented from interacting with the moderation code?

Reply:
When a guest visits a moderated code, simply reply Forbidden like as if they don't have the role. There would be no moderation whatsoever, because there's no way to notify the user about the moderation request. The test should use [email login](https://github.com/radiasoft/sirepo/blob/8cb1f59751554819c468bf4cdd36601c23a8637e/tests/auth/email1_test.py#L157) for creating the user. Bonus points to modularize it after copying it.

Reply:
I think user_name should always return something. For guests, the user_name could be `guest-<uid>`. Don't worry about auth.basic or auth.bluesky. auth.github would be fine (handle) and auth.email works, obviously. If we add ldap, it will probably be for ActiveDirectory so user names will be emails, too.

During the initial moderation check, if the user_name is not an email (use pyisemail.is_email), then we have no way of contacting them, and therefore, we can't moderate, i.e. just raise forbidden as if the role doesn't exist.

Reply:
ad61fc813610afe4d4c57cbe0e4322c503d6ac6c